### PR TITLE
Add support to list custom roles for organizations

### LINKED ISSUE #2327: Need a way to list custom roles for organizations
It would be great to have a way to list custom roles create in an organization.

### PR DESCRIPTION
**Pull request by tamboliasir1**
Fixes: #2327 .

Thanks for this opportunity.

### CODE REVIEW

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:1]: Please add the standard header to this file:

```
// Copyright 2022 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:8]: Please use complete sentences with punctuation for all godoc comments for auto-generated documentation.
Let's also add `Repo` to the name to better match the official documentation.

```suggestion
// OrginizationCustomRepoRoles represents custom repository roles available in specified organization.
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:20]: Idiomatic Go requires that the godoc comments start with the name of the method (and the spelling must match):

```suggestion
// ListCustomRepoRoles lists the custom repository roles available in this organization.
// In order to see custom repository roles in an organization, the authenticated user must be an organization owner.
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:9]:
```suggestion
type OrginizationCustomRepoRoles struct {
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:11]:
```suggestion
	CustomRepoRoles []*CustomRepoRoles `json:"custom_roles,omitempty"`
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:14]:
```suggestion
// CustomRepoRoles represents custom repository roles for an organization.
// See https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/managing-custom-repository-roles-for-an-organization
// for more information.
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:15]:
```suggestion
type CustomRepoRoles struct {
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:23]:
```suggestion
func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org string) (*OrginizationCustomRepoRoles, *Response, error) {
```

File: github/org_custom_roles_test.go
Comment (gmlewis) [inline github/org_custom_roles_test.go:1]:
```suggestion
// Copyright 2022 The go-github AUTHORS. All rights reserved.
//
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package github
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @tamboliasir1 !
Let's please change some of the naming and update the comments.

**Comment by gmlewis** [comment]
Please also make sure to run `go generate ./...` and push (not force-push) the results as described in our CONTRIBUTING.md document.

File: github/org_custom_roles.go
Comment (reedloden) [reply github/org_custom_roles.go:8]: Organization* here and elsewhere.

**Comment by tamboliasir1** [comment]
Hi @gmlewis I have made changes and pushed commit. Can you please review? Thank you so much.

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:30]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/orgs/custom-roles
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:31]: Were you able to test this endpoint out?  I ask because I don't think it will work as written.
According to the docs that I referenced above, it returns a list, not a single item.

```suggestion
func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org string) ([]*OrgainizationCustomRepoRoles, *Response, error) {
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:39]:
```suggestion
	var customRepoRoles []*OrgainizationCustomRepoRoles
```

File: github/org_custom_roles.go
Comment (gmlewis) [inline github/org_custom_roles.go:40]:
```suggestion
	resp, err := s.client.Do(ctx, req, &customRepoRoles)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/org_custom_roles.go
Comment (gmlewis) [reply github/org_custom_roles.go:31]: WHUPS!  I apologize... I misread the docs.  I think you are correct.  Please disregard.

File: github/org_custom_roles.go
Comment (gmlewis) [reply github/org_custom_roles.go:39]: Please disregard.

File: github/org_custom_roles.go
Comment (gmlewis) [reply github/org_custom_roles.go:40]: Please disregard.

**Review by gmlewis — APPROVED** [review]
Sorry, I was wrong.
After you fix the URL in the comments, LGTM.
We will then be ready for a second LGTM+Approval from any other contributor to this repo before merging.

File: github/github-accessors.go
Comment (reedloden) [inline github/github-accessors.go:9330]:
```suggestion
func (o *OrganizationCustomRepoRoles) GetTotalCount() int {
```

File: github/github-accessors_test.go
Comment (reedloden) [inline github/github-accessors_test.go:10932]:
```suggestion
func TestOrganizationCustomRepoRoles_GetTotalCount(tt *testing.T) {
```

File: github/github-accessors_test.go
Comment (reedloden) [inline github/github-accessors_test.go:10934]:
```suggestion
	o := &OrganizationCustomRepoRoles{TotalCount: &zeroValue}
```

File: github/github-accessors_test.go
Comment (reedloden) [inline github/github-accessors_test.go:10936]:
```suggestion
	o = &OrganizationCustomRepoRoles{}
```

File: github/org_custom_roles.go
Comment (reedloden) [inline github/org_custom_roles.go:13]:
```suggestion
// OrganizationCustomRepoRoles represents custom repository roles available in specified organization.
```

File: github/org_custom_roles.go
Comment (reedloden) [inline github/org_custom_roles.go:14]:
```suggestion
type OrganizationCustomRepoRoles struct {
```

File: github/org_custom_roles.go
Comment (reedloden) [inline github/org_custom_roles.go:31]:
```suggestion
func (s *OrganizationsService) ListCustomRepoRoles(ctx context.Context, org string) (*OrganizationCustomRepoRoles, *Response, error) {
```

File: github/org_custom_roles.go
Comment (reedloden) [inline github/org_custom_roles.go:39]:
```suggestion
	customRepoRoles := new(OrganizationCustomRepoRoles)
```

File: github/org_custom_roles_test.go
Comment (reedloden) [inline github/org_custom_roles_test.go:32]:
```suggestion
	want := &OrganizationCustomRepoRoles{TotalCount: Int(1), CustomRepoRoles: []*CustomRepoRoles{{ID: Int64(1), Name: String("Developer")}}}
```

**Review by reedloden — CHANGES_REQUESTED** [review]
Please fix the typos. :-)

**Comment by tamboliasir1** [comment]
HI @reedloden @gmlewis Sorry for this silly typo I have resolved it please review. Please let me know if there is anything. Thank you.

**Review by gmlewis — APPROVED** [review]
Thank you, @tamboliasir1 !
And thank you, @reedloden for catching the typos!
LGTM.

Awaiting second LGTM from any other contributor to this repo before merging.

**Review by joshuahancox — APPROVED** [review]
LGTM (Not a previous contributor though, just an interested party 😄 )

**Comment by tamboliasir1** [comment]
Thank you @gmlewis. Would love to contribute more!!😄

**Comment by reedloden** [comment]
@gmlewis can we get a release of `go-github` that includes this feature?

**Comment by gmlewis** [comment]
> @gmlewis can we get a release of `go-github` that includes this feature?

I will work on getting a release out before Saturday.

**Comment by gmlewis** [comment]
> @gmlewis can we get a release of go-github that includes this feature?

@reedloden - https://github.com/google/go-github/releases/tag/v44.0.0 is now available that contains this PR.

**Comment by reedloden** [comment]
> > @gmlewis can we get a release of go-github that includes this feature?
> 
> @reedloden - https://github.com/google/go-github/releases/tag/v44.0.0 is now available that contains this PR.

@gmlewis woot woot. Thank you so much!

**Pull request merged**
